processor/src/main/java/org/mapstruct/ap/internal/model/source/DefaultOptions.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.internal.model.source;
+
+import java.util.Collections;
+import java.util.Set;
+import javax.lang.model.type.DeclaredType;
+
+import org.mapstruct.ap.internal.option.Options;
+import org.mapstruct.ap.internal.prism.BuilderPrism;
+import org.mapstruct.ap.internal.prism.CollectionMappingStrategyPrism;
+import org.mapstruct.ap.internal.prism.InjectionStrategyPrism;
+import org.mapstruct.ap.internal.prism.MapperPrism;
+import org.mapstruct.ap.internal.prism.MappingInheritanceStrategyPrism;
+import org.mapstruct.ap.internal.prism.NullValueCheckStrategyPrism;
+import org.mapstruct.ap.internal.prism.NullValueMappingStrategyPrism;
+import org.mapstruct.ap.internal.prism.NullValuePropertyMappingStrategyPrism;
+import org.mapstruct.ap.internal.prism.ReportingPolicyPrism;
+
+public class DefaultOptions extends DelegatingOptions {
```

Comment:
What do `DefaultOptions` mean?

Reply:
When there's no Mapping, BeanMapping, Mapper, MapperConfig defined, in the end it falls back to what we defined as "default". 

We designed an hierarchy in annotations. Programmatically the question is whether this is correct on a more fundamental level. For instance, we define a `default` in the annotation for the `nullValuePropertyMappingStrategy` in  `@Mapping`, being `NullValuePropertyMappingStrategy nullValuePropertyMappingStrategy()
        default NullValuePropertyMappingStrategy.SET_TO_NULL`

We don't actually use this default, other than for convenience: the user does not have to specify this `nullValuePropertyMappingStrategy` in `@Mapping`. Instead of this, we check whether it was set explicitly or not. If not, we fall "through" the `@Mapping` to the next in line, the `@BeanMapping` were we repeat the same trick. Then to `@Mapper`, `@MappingConfig` `Options` (for some) and in the end.. `DefaultOptions`.. Its a chain-of-responsibility.

Reply:
Makes sense now, thanks